src/editor/hooks/useActionManager.ts

```
@@ -0,0 +1,88 @@
+// src/editor/hooks/useActionManager.ts
+import { useState } from "react";
+import type { ComponentEvent } from "../stores/component-config";
+import type { ActionConfig } from "../components/Setting/ComponentEvent/ActionModal";
+import type { Component } from "../stores/components";
+
+// 这个Hook封装了所有关于Action的CRUD操作和模态框状态
+export function useActionManager(
+  curComponent: Component,
+  updateComponentProps: (id: number, props: any) => void
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)